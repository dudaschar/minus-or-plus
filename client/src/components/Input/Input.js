import React from 'react';

import { BaseInput } from './styles/BaseInput';

function Input(props) {
  return (
    <BaseInput
      type="text"
      {...props} 
    />
  );
}


export default Input;
