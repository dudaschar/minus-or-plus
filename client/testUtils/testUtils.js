import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../stylesUtils/theme';

const TestingWithProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

TestingWithProviders.propTypes = {
  children: PropTypes.any.isRequired,
};

const customRender = (ui, options) => (
  render(ui,  { wrapper: TestingWithProviders, ...options })
);

export * from '@testing-library/react';
export { customRender as render };

