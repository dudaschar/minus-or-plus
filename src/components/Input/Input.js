import React, { useState } from 'react';

import { BaseInput } from './styles/BaseInput';

function Input(props) {
  const [number, setNumber] = useState();

  const handleNumberInput = ({ target }) => {
    const { value } = target;
    const resetValue = value === '' ? true : false;
    const inputValue = Number(value);

    if (resetValue) {
      return setNumber(value);
    }

    if (!inputValue) {
      return;
    }

    setNumber(inputValue);
  };

  return (
    <BaseInput
      {...props}
      value={number}
      onChange={handleNumberInput}  
    />
  );
}

export default Input;
