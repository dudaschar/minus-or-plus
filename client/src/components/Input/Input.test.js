import React from 'react';
import { render } from 'testUtils/testUtils';

import Input from './Input';

describe('Input Component', () => {
  it('should render properly', () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });
});