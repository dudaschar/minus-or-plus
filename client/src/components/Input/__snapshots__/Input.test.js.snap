// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render properly 1`] = `
<DocumentFragment>
  <input
    class="BaseInput-z5k0nq-0 hsPGhi"
    type="text"
  />
</DocumentFragment>
`;
